core/src/main/resources/error/error-classes.json

```diff
@@ -1,16 +1,50 @@
 {
+  "ALTER_TABLE_WITH_DROP_PARTITION_AND_PURGE_UNSUPPORTED_ERROR" : {
+    "message" : [ "ALTER TABLE ... DROP PARTITION ... PURGE" ],
+    "sqlState" : "42000"
+  },
   "AMBIGUOUS_FIELD_NAME" : {
     "message" : [ "Field name %s is ambiguous and has %s matching fields in the struct." ],
     "sqlState" : "42000"
   },
+  "CANNOT_FETCH_TABLES_OF_DATABASE_ERROR" : {
+    "message" : [ "Unable to fetch tables of db %s" ]
+  },
+  "CANNOT_RECOGNIZE_HIVE_TYPE_ERROR" : {
+    "message" : [ "Cannot recognize hive type string: %s, column: %s" ],
+    "sqlState" : "0D000"
+  },
+  "CANNOT_SET_TIMEOUT_DURATION_ERROR" : {
+    "message" : [ "Cannot set timeout duration without enabling processing time timeout in [map|flatMap]GroupsWithState" ],
+    "sqlState" : "42000"
+  },
+  "CONVERT_HIVE_TABLE_TO_CATALOG_TABLE_ERROR" : {
+    "message" : [ "%s, db: %s, table: %s" ],
+    "sqlState" : "42000"
+  },
   "DIVIDE_BY_ZERO" : {
     "message" : [ "divide by zero" ],
     "sqlState" : "22012"
   },
+  "DROP_TABLE_WITH_PURGE_UNSUPPORTED_ERROR" : {
+    "message" : [ "DROP TABLE ... PURGE" ],
+    "sqlState" : "42000"
+  },
   "DUPLICATE_KEY" : {
     "message" : [ "Found duplicate keys '%s'" ],
     "sqlState" : "23000"
   },
+  "FAILED_RENAME_TEMP_FILE_ERROR" : {
+    "message" : [ "Failed to rename temp file %s to %s as rename returned false" ],
+    "sqlState" : "42000"
+  },
+  "GET_PARTITION_METADATA_BY_FILTER_ERROR" : {
+    "message" : [ "Caught Hive MetaException attempting to get partition metadata by filter\nfrom Hive. You can set the Spark configuration setting\n%s to true to work around\nthis problem, however this will result in degraded performance. Please\nreport a bug: https://issues.apache.org/jira/browse/SPARK" ]
```

Comment:
I believe this error message is meant to be a single string without any newlines. Given how complex this is, however, we can turn this into an array of strings which will be joined with newlines for the user. In this case, it would look like:
```
"message" : [
  "Caught Hive MetaException attempting to get partition metadata by filter from Hive.",
  "You can set the Spark configuration setting %s to true to work around this problem, however this will result in degraded performance.",
  "Please report a bug: https://issues.apache.org/jira/browse/SPARK" ]
```